include/grpc++/impl/codegen/thrift_serializer.h

```diff
@@ -200,11 +200,9 @@ class ThriftSerializer {
   bool serialize_version_;
 
   void prepare() {
-
-    buffer_.reset(new TMemoryBuffer());
-
+    buffer_ = boost::make_shared<TMemoryBuffer>(*(new TMemoryBuffer()));
     // create a protocol for the memory buffer transport
-    protocol_.reset(new Protocol(buffer_));
+    protocol_ = std::make_shared<Protocol>(*(new Protocol(buffer_)));
```

Comment:
This defeats the purpose of `make_shared` . See my above comment; should just be `std::make_shared<Protocol>(buffer_)`